package/tvheadend-app-osmc/files/DEBIAN/postinst

```diff
@@ -41,6 +94,61 @@ if [ "$1" = "configure" ]; then
 		fi
 	fi
 
-	if ischroot; then exit 0; fi
+	# start tvh so that it updates any old user files
+	echo Starting tvheadend to finalise setup for osmc
 	if systemctl is-enabled tvheadend.service >/dev/null; then systemctl start tvheadend.service; fi
+	#systemctl restart tvheadend
+	sleep 5
```

Comment:
I thought I had an answer last night, but all along I've been testing with the default osmc setup. So tried setting up some user 'profiles' to see if those came across, and hit the buffers again.
Had a quick look at the tvh migration code in config.c and it would take some time for me to unravel, not least because it's 20 years since I last did any work in C, and that wasn't very complicated.  The tvh data format has changed no less than 8 times (v17 - v25) and the migration does 8 updates in turn.  I think one of my problems is if I write any settings in 4.2.2 format and you hit it with the v17-v18 migration code it doesn't understand.

I won't give up, though.